Tests/UnitTests/Plugins/GeoModel/GeoModelMuonMockupBluePrintTests.cpp

```diff
@@ -0,0 +1,236 @@
+  auto barrelNode = std::make_shared<StaticBlueprintNode>(
+      std::make_unique<Acts::TrackingVolume>(
+          Transform3::Identity(),
+          std::make_shared<CylinderVolumeBounds>(rmin, rmax, 4_m),
+          name + "_Barrel"));
+
+  // create the bluprint nodes for the chambers and add them as children to the
+  // cylinder barrel node
+  for (auto& chamber : volChambers) {
+    auto chamberNode =
+        std::make_shared<StaticBlueprintNode>(std::move(chamber));
+    barrelNode->addChild(std::move(chamberNode));
+  }
+
+  return barrelNode;
+}
+
+BOOST_AUTO_TEST_SUITE(GeoModelMuonMockupBluePrintTests)
+
+BOOST_AUTO_TEST_CASE(MockupMuonGen3Geometry) {
+  // Read the geomodel file to create the mockup geometry
+
+  const GeometryContext context;
+  auto logger = Acts::getDefaultLogger("UnitTests", Acts::Logging::VERBOSE);
+
+  Acts::GeoModelTree gmTree =
+      readFromDb("/eos/user/c/cimuonsw/GeometryFiles/DimitrasMockup.db");
+  std::vector<std::string> queries = {"Muon"};
+  std::vector<std::unique_ptr<TrackingVolume>> chamberVolumes;
+
+  Acts::GeoModelDetectorObjectFactory::Options options;
+  options.queries = queries;
+
+  // Create the detector object factory
+  Acts::GeoModelDetectorObjectFactory::Config cfg;
+  cfg.nameList = {"RPC", "Tube", "BIL", "BML", "BOL"};
+  cfg.materialList = {"RPCgas", "Aluminium"};
+  cfg.convertSubVolumes = true;
+  cfg.convertBox = {"MDT"};
+
+  Acts::GeoModelDetectorObjectFactory factory(cfg);
+  Acts::GeoModelDetectorObjectFactory::Cache cache;
+  factory.construct(cache, context, gmTree, options);
+  std::cout << "box size=" << cache.boundingBoxes.size() << std::endl;
+
+  auto innerBarrel =
+      buildBarrelNode(cache.boundingBoxes, cache.sensitiveSurfaces, "BIL",
+                      *logger->clone(std::nullopt, Logging::DEBUG));
+  auto middleBarrel =
+      buildBarrelNode(cache.boundingBoxes, cache.sensitiveSurfaces, "BML",
+                      *logger->clone(std::nullopt, Logging::DEBUG));
+  auto outerBarrel =
+      buildBarrelNode(cache.boundingBoxes, cache.sensitiveSurfaces, "BOL",
+                      *logger->clone(std::nullopt, Logging::DEBUG));
+
+  // Check the number of children in each barrel
+  BOOST_CHECK(innerBarrel->children().size() == nSectors);
+  BOOST_CHECK(middleBarrel->children().size() == nSectors);
+  BOOST_CHECK(outerBarrel->children().size() == nSectors);
+
+  // Create the blueprint
+  Blueprint::Config bpCfg;
+  bpCfg.envelope[AxisDirection::AxisZ] = {20_mm, 20_mm};
+  bpCfg.envelope[AxisDirection::AxisR] = {2_mm, 20_mm};
+  Blueprint root{bpCfg};
+  auto& cyl =
+      root.addCylinderContainer("CylinderContainer", AxisDirection::AxisR);
+  cyl.setAttachmentStrategy(VolumeAttachmentStrategy::Gap);
+
+  // Add the volumes to the blueprint
+  cyl.addChild(std::move(innerBarrel));
+  cyl.addChild(std::move(middleBarrel));
+  cyl.addChild(std::move(outerBarrel));
+
+  auto trackingGeometry = root.construct({}, context, *logger);
```

Comment:
You could add at least a basic check that the total number of volumes and the total number of surfaces is the expected one.